src/adagrams.js

```diff
@@ -1,15 +1,152 @@
+    'H': [2,4],
+    'I': [9,1],
+    'J': [1,8],
+    'K': [1,5],
+    'L': [4,1],
+    'M': [2,3],
+    'N': [6,1],
+    'O': [8,1],
+    'P': [2,3],
+    'Q': [1,10],
+    'R': [6,1],
+    'S': [4,1],
+    'T': [6,1],
+    'U': [4,1],
+    'V': [2,4],
+    'W': [2,4],
+    'X': [1,8],
+    'Y': [2,4],
+    'Z': [1,10]
+}
+
+const letterPool = []
+
+// Helper Function to Build Letter Pool
+const buildPool = () => {
+
+  for (let letter in letterObject) {
+      for (let i=0; i < letterObject[letter][0]; i++) {
+          letterPool.push(letter);
+      }
+  }
+
+  return letterPool;
+
+};
+
+buildPool();
+
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+
+  const shuffledList = letterPool.sort(() => 0.5 - Math.random());
+  return shuffledList.slice(0,10);
+
+};
+
+
+// Helper function to count num times a letter appears in string/array
+const countNum = (item) => {
+  
+  const numDict = {};
+  
+  for (let letter of item) {
+
+    letter = letter.toUpperCase()
+
+    if (numDict[letter]) {
+      numDict[letter] = numDict[letter] + 1;
+    } else {
+        numDict[letter] = 1;
+    }
+  }
+
+  return numDict;
+
 };
 
+
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  
+  const inputUpper = input.toUpperCase();
+  const numTimesInInput = countNum(input);
+  const numTimesInHand = countNum(lettersInHand);
+
+  for (let letter of inputUpper) {
+    if (lettersInHand.includes(letter)==false || (numTimesInInput[letter]!=numTimesInHand[letter])) {
+      return false;
+    } 
+  }
+
+  return true;
+
 };
 
+
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+  
+  const upperWord = word.toUpperCase();
+  let score = 0;
+
+  for (let letter of upperWord) {
+    score += letterObject[letter][1];
+  }
+
+  if (upperWord.length >= 7 && upperWord.length < 11) {
+    score += 8;
+  }
+
+  return score;
+
 };
 
+
+// Helper function to build object (word: wordScore)
+const buildWordScoreObject = (words) => {
+  const wordScores = {};
+
+  for (let word of words) {
+    wordScores[word] = scoreWord(word);
+  }
+
+  return wordScores;
+}
+
+
 export const highestScoreFrom = (words) => {
-  // Implement this method for wave 4
+  
+  const wordScoreObject = buildWordScoreObject(words);
+  const highScore = Math.max(...Object.values(wordScoreObject));
+  
+  let winningWord = {
+    'word': '',
+    'score': 0,
+  };
+  
+  for (let word of words) {
+    const calculatedScore = wordScoreObject[word];
+    
+    if (word.length == 10 && calculatedScore == highScore) {
+      winningWord['word'] = word;
+      winningWord['score'] = calculatedScore;
+      return winningWord;
+    }
+    
+    if ((winningWord['score'] < calculatedScore) || (winningWord['score'] == calculatedScore && winningWord['word'].length > word.length)) {
+      winningWord['word'] = word;
+      winningWord['score'] = calculatedScore;
+    }
+  }
+
+  return winningWord;
```

Comment:
Great job on translating your python adagrams into JS, Maz! It was very easy to follow and quite clever in some cases as well! Nice work on the helper functions as well! ⭐️